tests/topotests/lib/topogen.py

```diff
@@ -1185,6 +1223,57 @@ def stop(self, wait=True, assertOnError=True):
         return ""
 
 
+class TopoPMAcct(TopoHost):
+    "PMAcct server abstraction."
+    # pylint: disable=too-few-public-methods
+
+    PRIVATE_DIRS = [
+        "/etc/pmacct",
+        "/var/run/pmacct",
+        "/var/log",
+    ]
+
+    def __init__(self, tgen, name, **params):
+        """
+        PMAcct usually uses the following parameters:
+        * `ip`: the IP address (string) for the host interface
+        * `defaultRoute`: the default route that will be installed
+          (e.g. 'via 10.0.0.1')
+        """
+        params["privateDirs"] = self.PRIVATE_DIRS
+        super(TopoPMAcct, self).__init__(tgen, name, **params)
+
+    def __str__(self):
+        gear = super(TopoPMAcct, self).__str__()
+        gear += " TopoPMAcct<>".format()
+        return gear
+
+    def start(self, server_dir):
+        """
+        Start running PMAcct daemon:
+        * Copy all server_dir* folder contents into /etc/pmacct
+        * Run PMAcct daemon with configuration server_dir*/pmacctd.conf
+        """
+        self.run("mkdir -p /etc/pmacct")
```

Comment:
We have PRIVATE_DIRS, no need to hardcode directories two times.

Reply:
done!